packages/editor/src/components/local-autosave-monitor/index.js

```diff
@@ -150,20 +153,30 @@ function useAutosavePurge() {
 	const {
 		postId,
 		isDirty,
+		isAutosaving,
+		didError,
 	} = useSelect( ( select ) => ( {
 		postId: select( 'core/editor' ).getCurrentPostId(),
+		postType: select( 'core/editor' ).getCurrentPostType(),
 		isDirty: select( 'core/editor' ).isEditedPostDirty(),
+		isAutosaving: select( 'core/editor' ).isAutosavingPost(),
+		didError: select( 'core/editor' ).didPostSaveRequestFail(),
 	} ) );
 
 	const lastIsDirty = useRef( isDirty );
+	const lastIsAutosaving = useRef( isAutosaving );
 
 	useEffect( () => {
-		if ( lastIsDirty.current && ! isDirty ) {
-			window.sessionStorage.removeItem( postKey( postId ) );
+		if (
+			( lastIsDirty.current && ! isDirty ) ||
+			( lastIsAutosaving.current && ! isAutosaving && ! didError )
```

Comment:
Should this be the opposite (when the autosaves finishes instead of when the autosaves starts)?

Also, what's the reasoning for clearing when the post becomes dirty?

Reply:
> (when the autosaves finishes instead of when the autosaves starts)

Isn't that the case already? I guess `useRef`'s notation with `.current` is confusing to read.

> Also, what's the reasoning for clearing when the post becomes dirty?

Likewise, it clears when the post is flushed.